packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx

```diff
@@ -1756,6 +1833,90 @@ export class DataTable {
     ).scrollTop;
   };
 
+  private updateRowHeightForDescriptions = (
+    rowHeight: number,
+    cellContainer: Element
+  ) => {
+    cellContainer.setAttribute("data-row-height", rowHeight.toString());
+    cellContainer.setAttribute(
+      "style",
+      `${this.ROW_HEIGHT_CSS_VARIABLE}: ${rowHeight}px`
+    );
+  };
+
+  // Method to update the row heights on cells with descriptions and tooltip truncation
+  private updateCellHeightsWithDescriptions = () => {
+    const descriptions = this.el.shadowRoot.querySelectorAll(
+      this.CELL_DESCRIPTION_STRING
+    );
+
+    descriptions.forEach((description) => {
+      const cellContainer = description.closest(
+        `.${this.CELL_CONTAINER_WITH_DESCRIPTION_STRING}`
+      ) as HTMLElement;
+      const typography = cellContainer.querySelector(
+        this.IC_TYPOGRAPHY_STRING
+      ) as HTMLIcTypographyElement;
+
+      const descriptionMarginTop = window
+        .getComputedStyle(description)
+        .getPropertyValue("margin-top");
+
+      const descriptionHeight =
+        description.clientHeight + parseInt(descriptionMarginTop, 10);
+
+      if (this.globalRowHeight && this.globalRowHeight !== "auto") {
+        if (
+          !typography.textContent &&
+          descriptionHeight + this.DEFAULT_LINE_HEIGHT > this.globalRowHeight
+        ) {
+          this.updateRowHeightForDescriptions(descriptionHeight, cellContainer);
+        } else if (this.truncationPattern === this.TOOLTIP_STRING) {
+          if (
+            descriptionHeight + this.DEFAULT_LINE_HEIGHT >
+            this.globalRowHeight
+          ) {
+            const cellIcon = cellContainer?.querySelector(".icon");
+            if (deviceSizeMatches(577) && cellIcon) {
```

Comment:
could the 577 be a constant, as also seems to be used elsewhere in this file? - there are also IC_DEVICE_SIZES which can be used for these types of things